### PR TITLE
Add regression coverage for issue 43572

### PR DESCRIPTION
**Pull request by labkey-chrisj**
#### Rationale
Adds a new test to verify that LabKey respects fKDisplayColumn settings in metadata

#### Related Pull Requests
replaces https://github.com/LabKey/biologics/pull/1017

#### Changes
just the new test

### CODE REVIEW

File: src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+package org.labkey.test.tests;
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java]: Should go in a module-specific package.
```suggestion
package org.labkey.test.tests.experiment;
```

File: src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    // it would be nice if our
+    private void setFieldValues(String name, String comment, String amount, String ingredient)
+    {
+        if (name != null)   // for update, name field is disabled
+            setFormElement(Locator.name("quf_Name"), name);
+        setFormElement(Locator.name("quf_comment"), comment);
+        setFormElement(Locator.name("quf_amount"), amount);
+        setFormElement(Locator.name("quf_ingredient"), ingredient);
+        clickAndWait(Locator.lkButton("Submit"));
+    }
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java]: `UpdateQueryRowPage` (returned by `DataRegionTable.clickEditRow`) might work for inserting new rows also. If you try it and it works, consider updating `DataRegionTable. clickInsertNewRow` to return one.
```suggestion
    private void setFieldValues(String name, String comment, String amount, String ingredient)
    {
        var insertPage = new UpdateQueryRowPage(this);
        if (name != null)   // for update, name field is disabled
            insertPage.setField("Name", name);
        insertPage.setField("comment", comment);
        insertPage.setField("amount", amount);
        insertPage.setField("ingredient", ingredient);
        insertPage.submit();
    }
```

**Review by labkey-tchad — APPROVED** [review]
Tests should, generally, go in a module-specific package. Also, a suggestion for an improvement to `setFieldValues`

File: src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java (hunk 2 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java]: I considered creating a page class for an `InsertQueryRowPage` and updating `DataRegionTable.clickInsertNewRow()` to return that, but I figured that would be seen as too much feature creep.  Thanks for suggesting!
It turns out that UpdateQueryRowPage works just fine for the insert scenario!

**Pull request merged**